src/rules/noUnnecessaryCastRule.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.TypedRule {
+  /* tslint:disable:object-literal-sort-keys */
+  public static metadata: Lint.IRuleMetadata = {
+    ruleName: "no-unnecessary-cast",
+    description: `Warns if a cast does not change the type of an expression.`,
+    options: null,
+    optionsDescription: "Not configurable",
+    type: "typescript",
+    typescriptOnly: true,
+    requiresTypeInfo: true,
+  };
+  /* tslint:enable:object-literal-sort-keys */
+
+  public static FAILURE_STRING = "This cast is unnecessary.";
```

Comment:
Please be more descriptive in the failure string and explain that the cast doesn't change the type of the expression. Also, to be slightly pedantic here (well within the rights of a _linter_), this is more correctly referred to as a type _assertion_ in TS syntax, so let's refer to it as such (we have other rules that relate to type assertions). The name should be changed to `no-unnecessary-type-assertion`.

`FAILURE_STRING = "This assertion is unnecessary since it does not change the type of the expression."`